tests/run.sh

```diff
@@ -148,16 +148,16 @@ debugfs -w -R "rm /lib/systemd/systemd-networkd" core-image-full-cmdline-$MACHIN
 
 dd if=/dev/urandom of=broken_update.ext4 bs=10M count=5
 
-XDIST_ARGS="${XDIST_ARGS:--n ${XDIST_PARALLEL_ARG:-auto}}"
-MAX_FAIL_ARG="--maxfail=1"
+# Contains either the arguments to xdists, or '--maxfail=1', if xdist not found.
+EXTRA_TEST_ARGS=
 HTML_REPORT="--html=report.html --self-contained-html"
 
 if ! pip2 list |grep -e pytest-xdist >/dev/null 2>&1; then
-    XDIST_ARGS=""
+    EXTRA_TEST_ARGS="--maxfail=1"
     echo "WARNING: install pytest-xdist for running tests in parallel"
 else
     # run all tests when running in parallel
-    MAX_FAIL_ARG=""
+    EXTRA_TEST_ARGS="${XDIST_ARGS:--n ${XDIST_PARALLEL_ARG:-auto}}"
```

Comment:
I think we could further optimize by unifying `XDIST_ARGS` and  `XDIST_PARALLEL_ARG` (i.e. get the rid of one). See mender-qa comment [here](https://github.com/mendersoftware/mender-qa/blob/master/.gitlab-ci.yml#L46).